impl/ex/lib/snowflake.ex

```diff
@@ -0,0 +1,129 @@
+defmodule Pika.Snowflake do
+  import Bitwise
+  alias Pika.Utils
+  use GenServer
+
+  @moduledoc """
+  `Pika.Snowflake` holds the state, generates Snowflakes, and decodes Snowflakes.
+
+  `Pika.Snowflake` should be started under a `Supervisor` or `Application` before you start using
+  `Pika.gen/0` or `Pika.deconstruct/1`
+
+  ```elixir
+  defmodule MyApp.Application do
+    use Application
+
+    def start(_type, _args) do
+      children = [Pika.Snowflake]
+
+      Supervisor.start_link(children, strategy: :one_for_one)
+    end
+  end
+  ```
+
+  or manually in `iex`
+
+  ```elixir
+  iex(1)> Pika.Snowflake.start_link()
+  {:ok, #PID<0.190.0>}
+  ```
+
+  ## Custom epoch
+
+  You can start `Pika.Snowflake` with a custom epoch by passing it:
+
+  ```elixir
+  Pika.Snowflake.start_link(1_650_153_600_000)
+  ```
+  """
+
+  def start_link(epoch) when is_integer(epoch) do
+    GenServer.start_link(__MODULE__, {Utils.compute_node_id(), epoch, 0, 0}, name: __MODULE__)
+  end
+
+  def start_link([]) do
```

Comment:
You should also implement the case for `start_link/0` so your tests and benchmarks run correctly